core/src/impl/Kokkos_HBWSpace.cpp

```diff
@@ -115,7 +107,7 @@ void * HBWSpace::allocate( const size_t arg_alloc_size ) const
       // Over-allocate to and round up to guarantee proper alignment.
       size_t size_padded = arg_alloc_size + sizeof(void*) + alignment ;
 
-      void * alloc_ptr = memkind_malloc(MEMKIND_TYPE, size_padded );
+      void * alloc_ptr = malloc( size_padded );
```

Comment:
Was the idea to take over HBWSpace?  `m_alloc_mech` kinda lies here, right?

Reply:
@mhoemmen , yeah, in a productized implementation we'd either just use malloc, or steal whatever HostSpace does to use m_alloc_mech. Good catch

Reply:
Does anyone actually use HBWSpace for its originally intended purpose?

Reply:
As it is present in develop, HBWSpace is broken, I need to put together a PR for that (it literally won't build as-is). Odds are that'll be a separate PR though

Reply:
Should we rename it?  `DebugHostSpace`?

Reply:
@mhoemmen , I was planning to duplicate it. So we'd have HBWSpace (hopefully updated), and DebugHostSpace (perhaps with an alias like "Kokkos::Experimental::SimulatedCudaSpace"), but if we want to get rid of HBWSpace, renaming is a valid option as well

Reply:
Yeah we need to fix HBW Space. The problem is we don't really have a test for it running and in production Trinity is running in Cache mode so HBW Space doesn't exist anyway.

Reply:
@crtrott , then on that other PR to fix HBW (which is a low priority) what I might do is let us at least verify its syntax. One way to do that would be having a (default-off) compile-time switch which makes HBWSpace just malloc/free, and have CI which turns that switch on. That way we could avoid bugs like the current one. I don't know what our answer is for true HBW Space CI, but we could at least make sure that *if* HBW Space functions properly, our semantics for using it are correct.

Reply:
HBWSpace smells like the first step to a polymorphic allocator (which I would have found useful on occasion).